src/main/java/fi/aalto/cs/apluscourses/utils/CommonUtil.java

```diff
@@ -0,0 +1,32 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.IntFunction;
+import org.jetbrains.annotations.NotNull;
+
+public class CommonUtil {
+
+  private CommonUtil() {
+
+  }
+
+  /**
+   * Creates a new {@link List} of type {@link T} and size {@code length} populated with objects
```

Comment:
`{@link <T>}` this syntax should work better in JavaDoc

Reply:
Are you sure? Checkstyle google checks reports parse error when using `{@link <T>}`. 🤔

Reply:
there were warning pointing to this when I tried to generate JavaDoc. I've actually created a new issue to generate and publish JavaDocs #73